route/route.go

```diff
@@ -546,3 +702,104 @@ func unmarshal(r *http.Request, data io.Reader, v interface{}) error {
 		return jsoniter.NewDecoder(data).Decode(v)
 	}
 }
+
+func getAPIKeyAndDatasetFromMetadata(md metadata.MD) (apiKey string, dataset string) {
+	apiKey = getFirstValueFromMetadata(types.APIKeyHeader, md)
+	if apiKey == "" {
+		apiKey = getFirstValueFromMetadata(types.APIKeyHeaderShort, md)
+	}
+	dataset = getFirstValueFromMetadata(types.DatasetHeader, md)
+
+	return apiKey, dataset
+}
+
+// safeGetValueFromMeta returns the first value of a metadata entry using a
```

Comment:
The other helper functions are jealous that this one gets a comment and the rest don't. :P Also the method name is different. 

```suggestion
// getFirstValueFromMetadata returns the first value of a metadata entry using a
```

Reply:
It's copied from hound/shepherd 😄